### PR TITLE
Switched from using hardcoded ignore lists in audit_deps.py to fetching from a separate repo

### PR DESCRIPTION
**Pull request by wknapik**
Resolves https://github.com/brave/devops/issues/8021

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [x] New files have MPL-2.0 license header
- [x] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [x] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [x] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

No QA testing needed. See https://github.com/brave/brave-core/pull/14405#issuecomment-1203292524 for how this was manually tested.

### CODE REVIEW

File: .pylint2rc [hunk 1]
```diff
@@ -93,7 +93,7 @@ evaluation=10.0 - ((float(5 * error + warning + refactor + convention) / stateme
 [FORMAT]
 
 # Maximum number of characters on a single line.
-max-line-length=80
+max-line-length=100
```
Comment (wknapik) [inline .pylint2rc:96]: https://bravesoftware.slack.com/archives/C7VLGSR55/p1659417448741579

Comment (mihaiplesa) [reply .pylint2rc:96]: let's revert once you got a green build and no other pylint issues

Comment (wknapik) [reply .pylint2rc:96]: This is already green in CI. But if we revert, then any change to this file will produce 3 pylint warnings in any PR that touches it. I think this limit is worth reconsidering anyhow.

**Review by linhkikuchi — APPROVED** [review]

File: .pylint2rc (hunk 1 above)
Comment (mihaiplesa) [reply .pylint2rc:96]: to be honest I'd the comment to exclude the pylons rule for every longer line. this also gives us indication how often this occurs.

Comment (wknapik) [reply .pylint2rc:96]: We could also do that.

That said, I think this is bad for code readability. IMO there's a price to be paid for every line a human has to read to understand what's going on. We can only hold so much in our minds before we lose track of things and the comments aren't just automatically filtered out, they do take some amount of brain power/capacity to process. And those comments contribute nothing towards explaining what the code does, they're pure noise to a human, only serve to distract.

File: script/audit_deps.py [hunk 2]
```diff
@@ -14,43 +14,37 @@
...
 from deps_config import RUST_DEPS_PACKAGE_VERSION
 
+# pylint: disable=inconsistent-return-statements
+def get_remote_audit_config(
+        url = "https://raw.githubusercontent.com/brave/audit-config/main/config.json",
+        retry = 3):
+    """Fetch additional audit configuration"""
+    for i in range(retry):
+        try:
+            return json.loads(requests.get(url).text)
```
Comment (kim0) [inline script/audit_deps.py:28]: requests has `response.json()` which should be more idiomatic
https://requests.readthedocs.io/en/latest/user/quickstart/#json-response-content

**Review by kim0 — DISMISSED** [review]

**Review by thypon — APPROVED** [review]

**Review by fmarier — APPROVED** [review]
Looks good to me. The only thing I'd ask you to check if you haven't already is that the audit fails if GitHub can't be reached for some reason (for example if it's blackholed via `/etc/hosts`).

File: script/audit_deps.py (hunk 2 above)
Comment (wknapik) [reply script/audit_deps.py:28]: Thanks, good suggestion. Fixed.

**Comment by wknapik** [comment]
> Looks good to me. The only thing I'd ask you to check if you haven't already is that the audit fails if GitHub can't be reached for some reason (for example if it's blackholed via /etc/hosts).

Yep, with `0.0.0.0 raw.githubusercontent.com` in /etc/hosts, the script fails.

File: .pylint2rc (hunk 1 above)
Comment (wknapik) [reply .pylint2rc]: Ok, I'll leave this discussion for another PR. Fixed in this one via a script-wide ignore for the line-too-long check.

**Comment by wknapik** [comment]
This was tested by running locally via `./script/audit_deps.py --source_root="$PWD" --rustup_path="$PWD/build/rustup" --cargo_path="$PWD/build/rustup" --audit_dev_deps` and in CI. The list of ignored advisories is the same, as before the change in public builds. Tests with `0.0.0.0 raw.githubusercontent.com` in /etc/hosts resulted in script failure, as they should.

**Review by fmarier — APPROVED** [review]

**Pull request merged**